src/codeu/chat/common/User.java

```diff
@@ -48,6 +52,11 @@ public User read(InputStream in) throws IOException {
     }
   };
 
+  public Collection<Uuid> UserSet = new HashSet<Uuid>(64);
```

Comment:
You mention you figured out that both are needed -- can you explain your reasoning?